pyproject.toml

```diff
@@ -1,3 +1,95 @@
 [build-system]
 requires = ['setuptools', 'wheel']
 build-backend = 'setuptools.build_meta'
+
+[project]
+name = 'envisage'
+version = '7.0.0'
+description = 'Extensible application framework'
+readme = 'README.rst'
+requires-python = '>= 3.7'
+license = {file = 'LICENSE.txt'}
+authors = [{name = 'Enthought', email = 'info@enthought.com'}]
+keywords = ['extensible', 'plugin', 'application', 'framework']
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Intended Audience :: Developers',
+    'Intended Audience :: Science/Research',
+    'License :: OSI Approved :: BSD License',
+    'Operating System :: MacOS :: MacOS X',
+    'Operating System :: Microsoft :: Windows',
+    'Operating System :: POSIX :: Linux',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: Implementation :: CPython',
+    'Topic :: Scientific/Engineering',
+    'Topic :: Software Development',
+    'Topic :: Software Development :: Libraries',
+    'Topic :: Software Development :: User Interfaces',
+]
+dependencies = [
+    'apptools',
+    'pyface',
+    'setuptools',
+    'traits>=6.2',
+    'traitsui',
+    'importlib-resources>=1.1.0; python_version<"3.9"',
+]
+
+[project.entry-points.'envisage.plugins']
+'envisage.core' = 'envisage.core_plugin:CorePlugin'
+
+[project.entry-points.etsdemo_data]
+demo_examples = 'envisage.examples._etsdemo_info:info'
+
+[project.optional-dependencies]
+pyside6 = ['pyside6<6.4.0']
```

Comment:
I'm not totally convinced that this is useful. It may make more sense to remove it.

Reply:
Thinking a bit more: this _is_ useful in that it records a necessary version restriction on PySide6, and that restriction isn't recorded anywhere else. However, that version restriction really belongs in Pyface, not here - it's Pyface that's the source of the incompatibility with PySide 6.4. Moreover, the next release of Pyface will be removing that restriction.

I think we should assume that we won't release Envisage 7.0 until after the next release of Pyface. In that case, we can remove this section of the `pyproject.toml` right now (and in the meantime, while we're waiting for the Pyface release, we can modify the GitHub Actions workflow to work with a supported version of PySide6).

Reply:
Done in 0ea87cfa193cffa42b89eb04c2e3180e833a708a